src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -298,9 +308,7 @@ def _testcase_reproduces_in_revision(testcase,
         error_type=uworker_msg_pb2.ErrorType.PROGRESSION_BUILD_SETUP_ERROR)
 
   build_data = testcase_manager.check_for_bad_build(job_type, revision)
-  # TODO(https://github.com/google/clusterfuzz/issues/3008): Move this to
-  # postprocess.
-  testcase_manager.update_build_metadata(job_type, revision, build_data)
+  progression_task_output.build_data_list.append(build_data.proto)
```

Comment:
@letitz, @jonathanmetzman 
I unfortunately am not able to simply append build_data object here and have to pass the proto.
I think we should go ahead and use protos directly.  What do you think?

Reply:
I generally support moving towards simpler protos with more explicit conversion code where needed.